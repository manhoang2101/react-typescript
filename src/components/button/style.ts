import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    appButton: {
      "&:hover": {
        backgroundColor: "#fff",
        borderColor: "#fff",
        boxShadow: "none",
      },
      "&:active": {
        boxShadow: "none",
        backgroundColor: "#0062cc",
        borderColor: "#005cbf",
      },
      "&:focus": {
        boxShadow: "0 0 0 0.2rem rgba(0,123,255,.5)",
      },
    },
  });

export default style;
