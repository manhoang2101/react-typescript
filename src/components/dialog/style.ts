import { Theme, createStyles } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    form: {
      display: "flex",
      flexDirection: "column",
      margin: "auto",
      width: "fit-content",
    },
    formControl: {
      marginTop: theme.spacing(2),
      minWidth: 120,
    },
    formControlLabel: {
      marginTop: theme.spacing(1),
    },
    appBar: {
      position: "relative",
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
    title: {
      marginLeft: 0,
      flex: 1,
    },
    gutters: {
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
  });

export default style;
