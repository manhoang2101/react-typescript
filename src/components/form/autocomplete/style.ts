import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    TextField: {
      paddingRight: 25,
    },

    MuiAutocomplete: {
      input: {
        paddingRight: 25,
      },
    },
    KeyboardArrowDownIcon: {
      paddingLeft: 0,
      borderLeft: `1px solid #ccc`,
    },
  });

export default style;
