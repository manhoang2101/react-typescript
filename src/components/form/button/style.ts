import { createStyles } from "@material-ui/core";

const style = () => createStyles({});

export default style;
