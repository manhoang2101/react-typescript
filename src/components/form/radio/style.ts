import { createStyles, Theme } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    MuiFormHelperText: {
      marginLeft: 0,
    },
  });

export default style;
