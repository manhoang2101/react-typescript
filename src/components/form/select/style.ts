import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    formControl: {
      minWidth: 120,
    },
  });

export default style;
