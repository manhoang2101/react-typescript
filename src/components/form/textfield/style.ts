import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    AppTextField: {
      display: "block",
    },
  });

export default style;
