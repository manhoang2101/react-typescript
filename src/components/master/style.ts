import { createStyles, Theme, fade } from "@material-ui/core";
import { drawerWidth } from "./constants";

const style = (theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
    },
    // header styles
    grow: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      display: "none",
      [theme.breakpoints.up("lg")]: {
        display: "block",
      },
      color: theme.palette.common.white,
    },
    search: {
      position: "relative",
      borderRadius: theme.shape.borderRadius,
      backgroundColor: fade(theme.palette.common.white, 0.15),
      "&:hover": {
        backgroundColor: fade(theme.palette.common.white, 0.25),
      },
      marginRight: theme.spacing(2),
      marginLeft: 0,
      width: "100%",
      [theme.breakpoints.up("sm")]: {
        marginLeft: theme.spacing(3),
        width: "auto",
      },
    },
    searchIcon: {
      width: theme.spacing(7),
      height: "100%",
      position: "absolute",
      pointerEvents: "none",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
    },
    inputRoot: {
      color: "inherit",
    },
    inputInput: {
      padding: theme.spacing(1, 1, 1, 7),
      transition: theme.transitions.create("width"),
      width: "100%",
      [theme.breakpoints.up("md")]: {
        width: 200,
      },
    },
    sectionDesktop: {
      display: "none",
      [theme.breakpoints.up("md")]: {
        display: "flex",
      },
    },
    sectionMobile: {
      display: "flex",
      [theme.breakpoints.up("md")]: {
        display: "none",
      },
    },
    appBar: {
      transition: theme.transitions.create(["margin", "width"], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    appBarShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(["margin", "width"], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },

    // !header styles

    hide: {
      display: "none",
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
    },
    drawerPaper: {
      width: drawerWidth,
    },
    drawerHeader: {
      display: "flex",
      alignItems: "center",
      padding: theme.spacing(0, 1),
      // necessary for content to be below app bar
      ...theme.mixins.toolbar,
      justifyContent: "flex-end",
    },
  });
export default style;
