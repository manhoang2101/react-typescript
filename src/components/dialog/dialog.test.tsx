import {
  withStyles,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  Button,
} from "@material-ui/core";
import style from "./style";
import AppDialog from ".";
import { shallow, mount } from "enzyme";
import React from "react";
describe("<AppDialog />", () => {
  const Component = withStyles(style)(AppDialog);
  const handleClickClose = jest.fn();
  const dialogTitle = () => (
    <DialogTitle id="max-width-dialog-title">
      Optional sizes FullWidth
    </DialogTitle>
  );
  const dialogContent = () => (
    <DialogContent>
      <DialogContentText>
        You can set my maximum width and whether to adapt or not.
      </DialogContentText>
      <form></form>
    </DialogContent>
  );
  const dialogActions = () => (
    <DialogActions>
      <Button onClick={handleClickClose} color="primary">
        Close
      </Button>
    </DialogActions>
  );

  test("should show text of DialogContent fullScreen", () => {
    const handleClickClose = jest.fn();
    const dialogTitle = () => (
      <DialogTitle className="max-width-dialog-title">
        Optional sizes FullWidth
      </DialogTitle>
    );
    const dialogActions = () => (
      <DialogActions>
        <Button
          className={`data-test-click`}
          onClick={handleClickClose}
          color="primary"
        >
          Close
        </Button>
      </DialogActions>
    );
    const dialogContent = () => (
      <DialogContent>
        <DialogContentText className={`data-content-test-text`}>
          You can set my maximum width and whether to adapt or not.
        </DialogContentText>
        <form></form>
      </DialogContent>
    );
    const container = mount(
      <Component
        open={false}
        onClose={handleClickClose}
        dialogTitle={dialogTitle}
        dialogContent={dialogContent}
        dialogActions={dialogActions}
      />
    );
    const content = container.find(".data-content-test-text");
    expect(content.at(0).text()).toBe(
      "You can set my maximum width and whether to adapt or not."
    );
  });
  test("should show text of DialogContent on click onClose", () => {
    const handleClickClose = jest.fn();
    const dialogActions = () => (
      <DialogActions>
        <Button
          className={`data-test-click`}
          onClick={handleClickClose}
          color="primary"
        >
          Close
        </Button>
      </DialogActions>
    );
    const dialogContent = () => (
      <DialogContent>
        <DialogContentText className={`data-content-test-text`}>
          You can set my maximum width and whether to adapt or not.
        </DialogContentText>
        <form></form>
      </DialogContent>
    );
    const container = mount(
      <Component
        open={false}
        fullScreen
        onClose={handleClickClose}
        dialogContent={dialogContent}
        dialogActions={dialogActions}
        dialogTitle={null}
      />
    );
    const content = container.find(".data-content-test-text");
    expect(content.at(0).text()).toBe(
      "You can set my maximum width and whether to adapt or not."
    );
  });
  test("should show text of DialogContent", () => {
    const handleClickClose = jest.fn();
    const dialogActions = () => (
      <DialogActions>
        <Button
          className={`data-test-click`}
          onClick={handleClickClose}
          color="primary"
        >
          Close
        </Button>
      </DialogActions>
    );
    const dialogContent = () => (
      <DialogContent>
        <DialogContentText className={`data-content-test-text`}>
          You can set my maximum width and whether to adapt or not.
        </DialogContentText>
        <form></form>
      </DialogContent>
    );
    const container = mount(
      <Component
        open={false}
        fullScreen
        onClose={handleClickClose}
        dialogContent={dialogContent}
        dialogActions={dialogActions}
        dialogTitle={null}
      />
    );
    const content = container.find(".data-test-click");
    content.at(0).simulate("click");
    expect(handleClickClose).toBeCalled();
  });
});
