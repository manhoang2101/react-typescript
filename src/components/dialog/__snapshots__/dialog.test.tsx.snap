// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppDialog /> toMatchSnapshot() 1`] = `ShallowWrapper {}`;
